### PR TITLE
Allow dropping travis cache

### LINKED ISSUE #46: Failed: could not retrieve Scala 2.12.12: missing scala.tools.nsc.Global
https://travis-ci.com/github/akka/akka-grpc-quickstart-java.g8/jobs/448185033#L260

```
[error] [error] [launcher] could not retrieve Scala 2.12.12: missing scala.tools.nsc.Global
[error] x akka-grpc-java-seed / scripted 
[error]  Cause of test exception: {line 1}  Command failed: Remote sbt initialization failed
[error] java.lang.RuntimeException: Failed tests:
[error] 	akka-grpc-java-seed / scripted
[error] 
[error] 	at scala.sys.package$.error(package.scala:30)
[error] 	at sbt.scriptedtest.ScriptedRunner.reportErrors(ScriptedTests.scala:565)
[error] 	at sbt.scriptedtest.ScriptedRunner.runAll(ScriptedTests.scala:568)
[error] 	at sbt.scriptedtest.ScriptedRunner.run(ScriptedTests.scala:513)
[error] 	at sbt.scriptedtest.ScriptedRunner.run(ScriptedTests.scala:477)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error] 	at sbt.ScriptedPlugin$.$anonfun$scriptedTask$3(ScriptedPlugin.scala:193)
[error] 	at sbt.ScriptedPlugin$.$anonfun$scriptedTask$3$adapted(ScriptedPlugin.scala:178)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:281)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] 	at sbt.Execute.work(Execute.scala:290)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:281)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:834)
[error] (scripted) Failed tests:
[error] 	akka-grpc-java-seed / scripted
```

### PR DESCRIPTION
**Pull request by raboof**
to hopefully fix #46

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,15 @@ install: jabba install "$TRAVIS_JDK" && jabba use "$_" && java -Xmx32m -version
 
 jobs:
   include:
+    - stage: drop-travis-caches
+      script:
+        - rm -rf $HOME/.m2
+        - rm -rf $HOME/.ivy2/cache
+        - rm -rf $HOME/.sbt/boot
+        - rm -rf $HOME/.sbt/launchers
```
Comment (ignasi35) [inline .travis.yml:16]:
```suggestion
```

This is currently not cached. 😅

**Review by ignasi35 — APPROVED** [review]

**Comment by ignasi35** [comment]
I've added `RUN_DROP_TRAVIS_CACHES_STAGE=true` as a public (non secret) ENV VAR available for all branches in the setting on the TravisCI web UI. Then relaunched the (still) failing job.

**Comment by ignasi35** [comment]
I think the build plan for this PR is already committed and we won't get a different build with a clean cache unless we send another commit on the PR.

**Review by octonato — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -38,6 +47,8 @@ jobs:
 
 stages:
   - name: check
+  - name: drop-travis-caches
+    if: env(RUN_DROP_TRAVIS_CACHES_STAGE) = true
```
Comment (octonato) [inline .travis.yml:51]: The ordering is wrong, that's why it failed. 

first drop, then check

Comment (octonato) [inline .travis.yml:51]:
```suggestion
  - name: drop-travis-caches
    if: env(RUN_DROP_TRAVIS_CACHES_STAGE) = true
  - name: check  
```

**Comment by raboof** [comment]
the drop cache uploads a cache:

    PR.48/cache--linux-xenial-9632d10a9809d17a5eeb2d7d9cdc036bd97602605eaaebbae154a320ab036321--jdk-default--scala-2.13.1.tgz

... but the failing job fetches another cache:

    PR.48/cache--linux-xenial-4a082dd6108cb29866e89deb0edda918357050dffe53206091ab55b2442c3e90--jdk-default--scala-2.13.1.tgz

what's going on here? :/

**Comment by ignasi35** [comment]
> what's going on here? :/

travis gonna travis.

I'm tempted to merge this, get a clean cache on `master` and rebase open PRs so they rebuild on a clean cache.

**Comment by raboof** [comment]
> > what's going on here? :/
> 
> travis gonna travis.
> 
> I'm tempted to merge this, get a clean cache on `master` and rebase open PRs so they rebuild on a clean cache.

can't hurt I guess

**Comment by raboof** [comment]
> can't hurt I guess

it didn't hurt, but didn't help either: also on master the jdk11 check job doesn't pick up the cache emptied by the `drop-travis-caches` stage...

**Comment by raboof** [comment]
Ah, https://docs.travis-ci.com/user/caching/:

> If a branch does not have its own cache, Travis CI fetches the cache of the repository’s default branch

and:

> There is one cache per branch and language version/ compiler version/ JDK version/ Gemfile location/ etc

... so we need to create a separate `drop-travis-caches` for jdk11 and merge that to master. will do.

**Pull request merged**